src/hb-ot-shape-complex-hangul.cc

```diff
@@ -214,7 +214,8 @@ preprocess_text_hangul (const hb_ot_shape_plan_t *plan HB_UNUSED,
       else
       {
 	/* No valid syllable as base for tone mark; try to insert dotted circle. */
-	if (font->has_glyph (0x25CCu))
+      if (   !(buffer->flags & HB_BUFFER_FLAG_DO_NOT_INSERT_DOTTED_CIRCLE)
+          && font->has_glyph (0x25CCu))
```

Comment:
Very minor, please use one tab instead 8 character as the current project style and remove extra spaces.

Some testing will be nice also. Thanks

Reply:
Can you please point me to some other test I could use as a model? I have not found a test for the other HB_BUFFER_FLAG.

Reply:
You can create an API test for it and write bare minimum check code to test with/without the flag result but I guess providing someway to have that from hb-view/hb-shape is the thing we may had.